### ISSUE TITLE
props-name-convention: a utility type such as `Omit<BucketProps, "bucketName">` is reported as "Props interface name 'Omit'"

### ISSUE THREAD

**Issue by ren-yamanashi** [ren-yamanashi/eslint-cdk-plugin#609]
### Describe the bug

`props-name-convention` takes the identifier of the props type reference as "the props interface name". When the props parameter is typed with a built-in utility type such as `Omit<BucketProps, "bucketName">` (a common shape for thin wrappers), the report says that the interface **`Omit`** should be renamed, which is not actionable.

`src/my-construct.ts`:

```ts
import { Bucket, BucketProps } from "aws-cdk-lib/aws-s3";
import { Construct } from "constructs";

export class MyConstruct extends Construct {
  constructor(scope: Construct, id: string, props: Omit<BucketProps, "bucketName">) {
    super(scope, id);
    new Bucket(this, "Bucket", { ...props, bucketName: "fixed-name" });
  }
}
```

Rule enabled: `awscdk/props-name-convention: "error"`.

```
$ npx eslint src/my-construct.ts
  5:52  error  Props interface name 'Omit' should follow '${ConstructName}Props' format. Expected 'MyConstructProps'  awscdk/props-name-convention

$ npx oxlint --format=unix -c .oxlintrc.json src/my-construct.ts
src/my-construct.ts:5:52: Props interface name 'Omit' should follow '${ConstructName}Props' format. Expected 'MyConstructProps'. [Error/awscdk(props-name-convention)]
```

`Partial<MyConstructProps>`, `Pick<...>`, `Readonly<...>` behave the same (`'Partial'`, `'Pick'`, `'Readonly'`).

### Expected Behavior

One of:

- a message that describes the actual situation, e.g. "Props of 'MyConstruct' should be declared as an interface named 'MyConstructProps'" (the rule can still report, since the convention is not met), or
- no report for TypeScript built-in utility types, consistent with unions / intersections / inline type literals, which the rule already skips.

Which one is the maintainer's call; the point of this issue is that `'Omit'` must not be presented as the interface name.

### Current Behavior

The utility type name is reported as the props interface name in both plugins.

### Plugin

eslint-plugin-awscdk, oxlint-plugin-awscdk

### Plugin Version

eslint-plugin-awscdk 4.4.2 / oxlint-plugin-awscdk 1.0.2

### Language Version

TypeScript 6.0.3 (eslint) / TypeScript 7.0.2 via oxlint-tsgolint 7.0.2001 (oxlint)

### Additional Context

```yaml
# machine-readable summary
rules: ["props-name-convention"]
symptom: ["misleading-message"]
affected_packages: ["eslint-plugin-awscdk", "oxlint-plugin-awscdk"]
root_cause: "packages/*/src/rules/props-name-convention.ts:49-53 (a TSTypeReference's typeName is used as the interface name regardless of type arguments or where the type is declared)"
verified_on: "main @ 9caea55 (eslint-plugin-awscdk 4.4.2, oxlint-plugin-awscdk 1.0.2)"
related_issues: []
```

A generic **user** props type (`props: MyConstructProps<T>`) is handled correctly today, so the distinction cannot be "has type arguments"; it is "is a TypeScript built-in utility type" (declared in a default lib file, or matched by name).

Reproduction environment:

- macOS 14.1, Node.js v24.18.0, pnpm 10.29.3
- aws-cdk-lib 2.268.0 / constructs 10.8.1
- ESLint 10.9.1 + typescript-eslint 8.69.0 + TypeScript 6.0.3
- oxlint 1.81.0 + oxlint-tsgolint 7.0.2001 (corsa-oxlint 1.13.1)
